### PR TITLE
Define emulator context for MelonDS (and add a 64-bit build configuration)

### LINKED ISSUE #284: x64 build
I would like to be able to link with x64 build of emulators.

This requires an x64 build configuration for `RAIntegration` and the ability to fetch the correct build from the server on update.

### PR DESCRIPTION
**Pull request by rzumer**
If you want it.

Adds an x64 build configuration (and closes #284), because I couldn't manage to compile melonDS in 32-bit mode without it failing to run on my PC, and apparently building it in 32-bit has a significant performance hit anyway.

The emulator is [there](https://github.com/rzumer/melonDS/tree/cheevos) and I will transfer ownership to RA if/when this gets merged. I did not go through the checklist yet but I can't think of anything missing off the top of my head, except for overlay navigation key binds being hardcoded for reasons.

And the justification for a standalone emulator is:
- High performance compared to RALibretro
- Many settings of MelonDS are not exposed in its libretro core
- Complete microphone support
- Wi-Fi support
- Screen rotation is much less painful to use than in RetroArch

~I thought it also had partial GBA slot support, but apparently not.~ Now implemented (https://github.com/Arisotura/melonDS/pull/550).

To build, follow instructions in the readme, but add `-DBUILD_RETROACHIEVEMENTS` to the `cmake` call and finish with `../ra-dist.sh` to copy the overlay folder.

### CODE REVIEW

File: tests/RA_Interface.Tests.vcxproj [hunk 1]
```diff
@@ -42,20 +65,32 @@
...
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+  </ImportGroup>
   <PropertyGroup Label="UserMacros" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
     <LinkIncremental>true</LinkIncremental>
     <OutDir>$(SolutionDir)bin\$(Configuration)\tests\RA_Interface\</OutDir>
     <IntDir>$(SolutionDir)obj\$(Configuration)\tests\RA_Interface\</IntDir>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
+    <LinkIncremental>true</LinkIncremental>
```
[the remark this answers is not in the document]
Comment (rzumer) [reply tests/RA_Interface.Tests.vcxproj]: Fixed.

File: tests/RA_Interface.Tests.vcxproj [hunk 2]
```diff
@@ -42,20 +65,32 @@
...
+  </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
     <LinkIncremental>false</LinkIncremental>
     <OutDir>$(SolutionDir)bin\$(Configuration)\tests\RA_Interface\</OutDir>
     <IntDir>$(SolutionDir)obj\$(Configuration)\tests\RA_Interface\</IntDir>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
+    <LinkIncremental>false</LinkIncremental>
```
[the remark this answers is not in the document]
Comment (rzumer) [reply tests/RA_Interface.Tests.vcxproj]: Fixed.

File: tests/RA_Interface.Tests.vcxproj [hunk 3]
```diff
@@ -78,6 +113,30 @@
...
+    <Link>
+      <SubSystem>Windows</SubSystem>
+      <AdditionalLibraryDirectories>$(VCInstallDir)UnitTest\lib;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+      <GenerateDebugInformation>DebugFull</GenerateDebugInformation>
+      <AdditionalDependencies>kernel32.lib;user32.lib;Winhttp.lib;%(AdditionalDependencies)</AdditionalDependencies>
+    </Link>
+  </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
     <ClCompile>
       <PrecompiledHeader>Use</PrecompiledHeader>
```
[the remark this answers is not in the document]
Comment (rzumer) [reply tests/RA_Interface.Tests.vcxproj]: The Win32 config set `NDEBUG` in release mode, so I did the same for the x64 config. Rest is done.

**Comment by rzumer** [comment]
> 
> 
> please move the `rcheevos` and `tests` project outputs to the `$(Platform)` subdirectories as well.

And done.

**Comment by Jamiras** [comment]
I had to make a few more minor changes to get this to compile.
[patch.diff.txt](https://github.com/RetroAchievements/RAIntegration/files/3986291/patch.diff.txt)

**Comment by rzumer** [comment]
> patch.diff.txt

Applied.

**Pull request merged**